api/server/errorhandler.go

```diff
@@ -17,18 +21,52 @@ func makeErrorHandler(err error) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		statusCode := httpstatus.FromError(err)
 		vars := mux.Vars(r)
-		if apiVersionSupportsJSONErrors(vars["version"]) {
-			response := &types.ErrorResponse{
-				Message: err.Error(),
-			}
+		version := vars["version"]
+
+		errMsg := err.Error()
+		stringErrs := unwrapErrors(err, true)
+		if len(stringErrs) > 1 {
+			errMsg = fmt.Sprintf("%d errors occurred:\n\t* %s", len(stringErrs), strings.Join(stringErrs, "\n\t* "))
+		}
+
+		structuredErrs := make([]*types.StructuredError, 0, len(stringErrs))
+		for _, stringErr := range stringErrs {
+			structuredErrs = append(structuredErrs, &types.StructuredError{Message: stringErr})
+		}
+
+		const firstAPIVersionWithJSONErrors = "1.24"
+		const firstAPIVersionWithJSONMultiErrors = "1.44"
+
+		switch {
+		case version == "" || versions.GreaterThanOrEqualTo(version, firstAPIVersionWithJSONMultiErrors):
+			response := &types.ErrorResponse{Message: errMsg, Errors: structuredErrs}
+			_ = httputils.WriteJSON(w, statusCode, response)
```

Comment:
Is it necessary to gate multi-errors on API version? We're adding a new property to the error response JSON document, not changing the semantics of any existing ones.

Reply:
Not sure if that's needed, I can drop it we consider it's not needed.